### ISSUE TITLE
TeraStation™ WSH5610DNS6 support

### ISSUE THREAD

**Issue by danielzx2** [1000001101000/Debian_on_Intel_Terastations#13]
I saw the previous thread from a few years ago about support for TeraStation™ WSH5610DNS6. Is it ongoing/complete? I'm thinking of installing Linux of somekind in order to turn into a media server. 

Also, is there a written guide of some kind for installing?

Thank you!

**Comment by 1000001101000** [comment]
Good to hear from you!

I finally got my hands on a WSH5610DN a while back thinking that it was either similar to either the older (TS-QVL/TS5400) or newer (WS5420) Intel Terastations. It turns out to be a completely different system that uses a hardware raid controller rather than being like any of the other Terastation models. 

I was able to get the Buffalo firmware stuff working on a generic Windows Server install, at least enough to get the RAID configuration utility working.... unfortunately I haven't gotten around to doing it again and taking notes. I started playing around with reverse engineering the protocol used to talk to the RAID controller (I think I remember thinking they were ata commands) .  

I was expecting it to be higher performance and more fun to work with than it turned out to be.... I kinda moved on to other stuff since I'm not planning to do much with mine and they are rarer than all the other models as far as I can tell.

I did run Debian on mine and it seemed to work fine. The RAID controller shows up as a generic SATA device with a single large disk that represents the array. It didn't provide any kind of passthru to see the underlying disks or state of the array if I remember correctly. I think I saw some commands to query SMART state of the drives as part of the custom Buffalo stuff but I could be remembering wrong. 

In any case, you can probably run linux on it but would need to boot stock Windows to configure the RAID array and would need to rely on the LCD screen for other information about the array. 

Talking about this does make it sound like something interesting to pull off the shelf and work on again (it always helps to know at least one other person is interested in something). There's a decent chance I'll work on this device some more some day.

**Comment by danielzx2** [comment]
I appreciate your expansive and that you still show interest in pursuing it.

I checked the backside of my station and was mistaken on my model number. It is actually a TS5400 Series. I saw that it was supported via your README so is it just to use the installer image as any regular OS installation?

**Comment by 1000001101000** [comment]
lol.

Yes, I have a note at the bottom of the readme with some basic instructions.

The installer image is a normal debian net installer image with some customizations for the Terastation hardware and also enables installing over SSH in case you don’t want a monitor/etc hooked up.

**Comment by danielzx2** [comment]
Perfect! Was thinking of ordering a VGA to HDMI since I don't have a VGQ monitor lying around but using SSH clears that problem.

Gonna look into it more then and see what happens. My goal is turning into a media server using Jellyfin.

**Comment by danielzx2** [comment]
Looking into it a bit, and I have one question.

What is the difference between Bullseye and Bookworm images in the installer-image directory? A bit confused on which one to pick for the SSH installation.

**Comment by 1000001101000** [comment]
Bookworm is the name of Debian 12, Bullseye is the name for Debian 11

**Comment by danielzx2** [comment]
Is anyone recommended for my model or does whichever work?

**Comment by 1000001101000** [comment]
You should use the latest version (Bookworm) unless you have a specific reason not too.

**Comment by danielzx2** [comment]
Okay thank you! Want to use primarily for Jellyfin and it does not require a specific version from what I've read.

**Comment by danielzx2** [comment]
Back again!

Was going through the installation today. Went through some hurdles but managed get passed except for one. It has to do with the GRUB boot loader. I cannot load it into any of my harddrives and I assume it's required for completing the installation. I used the thumbstick I used to write the image for now.

**Comment by 1000001101000** [comment]
If you’re using GPT and RAID (might be either or) you need to create a small (8mb) partition and mark it as type EF02 (I think) which it uses to load the extra drivers etc needed to boot from those volumes.

**Comment by danielzx2** [comment]
It already completed writing the partition to the thumbstick. Is it possible to create partition post-installation? Or is there a way to restart the installation from the beginning?

**Comment by 1000001101000** [comment]
depending on your layout you can probably add the the EF02 to each of the disks somewhere and then trigger a grub-install

I think I included a “update-boot” script that partially automates the process by searching for drives that contain the root volume and trying to install grub to them whenever a new initramfs is generated

**Comment by danielzx2** [comment]
Going to look into it more later today! Thinking of creating a empty partitions on all my disks and making a RAID (there is an old RAID5 currently on it but it's empty) and then see what I can do from there. Still at the installation step.

**Comment by danielzx2** [comment]
Got it to work! Shows as Debian 12.8.

I can SSH to it as well!

Anything else I should do post-installation before I start installing things?

**Comment by 1000001101000** [comment]
Make sure you generate a new initramfs and test a reboot after making any changes to disks/partitions/filesystems. Best to find any problems right away rather than in a few months after you’ve made a bunch of changes

**Comment by danielzx2** [comment]
Are there any special commands for this installation? Or can I use the standard method?

**Comment by 1000001101000** [comment]
Not sure I understand what you’re asking or what you’re trying to do at this phase.

There isn’t anything special about the Debian install beyond the added components listed in the readme

**Comment by danielzx2** [comment]
Sorry, I mean for initramfs. Is there any special commands to generate it for this installation?

**Comment by 1000001101000** [comment]
just the normal update-initramfs commands

**Comment by danielzx2** [comment]
Hi again! It worked with initramfs and created a RAID array with 3 drives. I have a couple of questions.

1. Is it possible to create with all 4 drives, including the boot drive?
2. Have you been able to play 4K media through the Buffalo?

**Comment by 1000001101000** [comment]
Yeah. You can even install the bootloader on all the drives so it can boot if drive(s) are missing 

I haven’t done streaming stuff from one of these.

**Comment by danielzx2** [comment]
Okay, will look into it 

Also, 4K streaming worked without issue, as long as you have the correct format
